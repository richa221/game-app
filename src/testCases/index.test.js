import reducer, {initialState}   from '../reducers/index';

describe('Index reducer Test', () => {          
    it('should return the initial state', () => {
      expect(reducer(undefined, {})).toMatchSnapshot();
    });

    it('should handle Default Request', () => {
      expect(
        reducer(initialState,
          {
            type: '',
          }),
      ).toMatchSnapshot();
    });

    it('should handle Default Req.', () => {
      const initialGameState = {
        gameState: {
          squares: [{"color": "red",
              "_color": "orange",
              "height": 80,
              "position": 0,
              "width": 80,
              "x": 0,
              "y": 0}]
        },
        currentCount: 0,
        maxMoves: 25,
      };
      expect(
        reducer(initialGameState,
          {
            type: 'SQUARE_CLICKED',
          }),
      ).toMatchSnapshot();
    });

});

