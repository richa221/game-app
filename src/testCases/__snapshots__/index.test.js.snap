// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index reducer Test should handle Default Req. 1`] = `
Object {
  "currentCount": 0,
  "gameState": Object {
    "squares": Array [
      Object {
        "_color": "orange",
        "color": "red",
        "height": 80,
        "position": 0,
        "width": 80,
        "x": 0,
        "y": 0,
      },
    ],
  },
  "maxMoves": 25,
}
`;

exports[`Index reducer Test should handle Default Request 1`] = `
Object {
  "gameState": Object {
    "currentCount": 0,
    "maxMoves": 25,
    "squares": Array [],
  },
}
`;

exports[`Index reducer Test should return the initial state 1`] = `
Object {
  "gameState": Object {
    "currentCount": 0,
    "maxMoves": 25,
    "squares": Array [],
  },
}
`;
